### PR TITLE
A few small tweaks and bug fixes

### PR DESCRIPTION
**Pull request by legokidlogan**
- Fixes powerups spawning slightly too high and needing players to jump into them
- Changes Grenadier to be ammo-based
- Attaches the ammo amount for Magnetic Crossbow to a config variable
- Fixes a clientside null entity error with Thorns

### CODE REVIEW

File: moon/powerups/client/thorns.moon [hunk 1]
```diff
@@ -106,8 +106,9 @@ net.Receive "CFC_Powerups-ThornsDamage", ->
...
-            thorn = Thorn ply, attacker, amount
-            manager\addThorn thorn
+            if IsValid( ply ) and IsValid( attacker )
+                thorn = Thorn ply, attacker, amount
+                manager\addThorn thorn
 
-            -- Hard limit the amount of beams rendered by the client
-            return if #manager.thorns >= 25
+                -- Hard limit the amount of beams rendered by the client
+                return if #manager.thorns >= 25
```
Comment (sarahsturgeon) [inline moon/powerups/client/thorns.moon]:
```
    for ply, attackers in pairs damageData do
        continue unless IsValid ply
        for attacker, amount in pairs attackers
            continue unless IsValid attacker
            thorn = Thorn ply, attacker, amount
            manager\addThorn thorn

            -- Hard limit the amount of beams rendered by the client
            return if #manager.thorns >= 25
```

File: moon/powerups/server/grenadier.moon [hunk 2]
```diff
@@ -17,9 +17,7 @@ class GrenadierPowerup extends BasePowerup
     new: (ply) =>
         super ply
 
-        duration = getConf "grenadier_duration"
-        @durationTimer = "CFC_Powerups-Grenadier-#{ply\SteamID64!}"
-        timer.Create @durationTimer, duration, 1, -> @Remove!
+        @UsesRemaining = getConf "grenadier_uses"
```
Comment (sarahsturgeon) [inline moon/powerups/server/grenadier.moon:20]:
```suggestion
        @usesRemaining = getConf "grenadier_uses"
```

and usage of

File: moon/powerups/server/magnetic_crossbow.moon [hunk 3]
```diff
@@ -163,7 +163,7 @@ class MagneticCrossbowPowerup extends BasePowerup
         super ply
 
         @PowerupHookName = "CFC-Powerups_MagneticCrossbow-#{@owner\SteamID64!}"
-        @UsesRemaining = 10
+        @UsesRemaining = getConf "crossbow_uses"
```
Comment (sarahsturgeon) [inline moon/powerups/server/magnetic_crossbow.moon:166]:
```suggestion
        @usesRemaining = getConf "crossbow_uses"
```

**Review by sarahsturgeon — CHANGES_REQUESTED** [review]

File: moon/powerups/client/thorns.moon [hunk 4]
```diff
@@ -105,9 +105,11 @@ net.Receive "CFC_Powerups-ThornsDamage", ->
     PrintTable damageData
 
     for ply, attackers in pairs damageData do
+        continue unless IsValid ply
```
Comment (sarahsturgeon) [inline moon/powerups/client/thorns.moon:108]:
```suggestion
        continue unless IsValid ply

```

File: moon/powerups/server/grenadier.moon [hunk 5]
```diff
@@ -17,9 +17,7 @@ class GrenadierPowerup extends BasePowerup
     new: (ply) =>
         super ply
 
-        duration = getConf "grenadier_duration"
-        @durationTimer = "CFC_Powerups-Grenadier-#{ply\SteamID64!}"
-        timer.Create @durationTimer, duration, 1, -> @Remove!
+        @usesRemaining = getConf "grenadier_uses"
```
Comment (sarahsturgeon) [inline moon/powerups/server/grenadier.moon]: You have mixed casing here. Is it `usesRemaining` or `UsesRemaining`

**Review by sarahsturgeon — CHANGES_REQUESTED** [review]

File: moon/powerups/server/grenadier.moon (hunk 5 above)
Comment (legokidlogan) [reply moon/powerups/server/grenadier.moon:20]: So does `magnetic_crossbow.moon`, the mixed casing came from ctrl-c'ing your code, so I'll fix it for both

File: moon/powerups/server/grenadier.moon [hunk 6]
```diff
@@ -88,6 +89,9 @@ class GrenadierPowerup extends BasePowerup
                     cluster\SetPos entPos
                     cluster\SetVelocity (newPos - entPos) * 5
 
+            if @UsesRemaining < 1
+                @Remove!
```
Comment (sarahsturgeon) [inline moon/powerups/server/grenadier.moon]:
```suggestion
            @Remove! if @UsesRemaining < 1
```

**Review by sarahsturgeon — APPROVED** [review]

**Pull request merged**